catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/mappings/CswRecordMapperFilterVisitor.java

```diff
@@ -176,10 +187,19 @@ public Object visit(PropertyIsEqualTo filter, Object extraData) {
             sourceIds.add((String) ((Literal) filter.getExpression2()).getValue());
             return null;
         }
+        AttributeType type =
+                attributeTypes.get(((PropertyName) filter.getExpression1()).getPropertyName());
+        if (type != null) {
+            type.getBinding()
+                    .cast(((Literal) filter.getExpression2()).getValue());
+        }
+
         Expression expr1 = visit(filter.getExpression1(), extraData);
         Expression expr2 = visit(filter.getExpression2(), expr1);
         boolean matchCase = filter.isMatchingCase();
-        return getFactory(extraData).equal(expr1, expr2, matchCase);
+        return ((Literal) (filter.getExpression2())).getValue() instanceof String ? getFactory(
```

Comment:
This ternary is just ugly. Sometimes it's better to be explicit with your if/else.